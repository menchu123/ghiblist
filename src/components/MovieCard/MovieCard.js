import "./MovieCard.scss";

const MovieCard = ({ movie }) => {
  return (
    <li className="movie">
      <section className="movie__image">
        <img
          className="movie__poster"
          src={movie.image}
          alt={`${movie.title} movie poster`}
          height="300"
        />
      </section>
      <section className="movie__info">
        <h3 className="movie__title">{movie.title}</h3>
        <div className="movie__data">
          <p className="movie__year">{movie.release_date}</p>
          <p className="movie__runtime">{movie.running_time}min</p>
          <p className="movie__score">{movie.rt_score}%</p>
        </div>
      </section>
    </li>
  );
};

export default MovieCard;
