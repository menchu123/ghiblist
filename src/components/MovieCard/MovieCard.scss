.movie {
  margin: 0;
  &__poster {
    height: 300px;
  }
  &__data {
    display: flex;
    font-size: 14px;
    width: 200px;
    justify-content: space-between;
    padding-right: 25px;
    margin-top: 5px;
    font-style: italic;
    border-bottom: 1px solid transparent;
  }
  &__title {
    font-size: 16px;
    font-weight: 700;
    border-bottom: 1px solid transparent;
  }
  &:hover {
    h3,
    .movie__data {
      border-bottom: 1px solid #3f37c9;
    }
    cursor: pointer;
  }
}
