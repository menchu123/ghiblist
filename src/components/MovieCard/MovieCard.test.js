import { render, screen } from "@testing-library/react";
import ReactTestRenderer from "react-test-renderer";
import MovieCard from "./MovieCard";

describe("Given a MovieCard component", () => {
  describe("When it is passed a movie object with the title 'Spirited Away'", () => {
    test("Then it should render a card with the movie's title", () => {
      const expectedName = "Spirited Away";
      const movie = {
        title: expectedName,
        director: "Hayao Miyazaki",
        producer: "Isao Takahata",
        release_date: "1986",
        running_time: "124",
        rt_score: "95",
      };

      render(<MovieCard movie={movie} />);

      const movieCard = screen.getByRole("listitem");
      const movieTitle = screen.getByRole("heading");

      expect(movieCard).toBeInTheDocument();
      expect(movieTitle.textContent).toBe(expectedName);
    });
  });
  describe("When it is passed a movie object", () => {
    test("Then it should render a card matching the snapshot", () => {
      const movie = {
        title: "Spirited Away",
        director: "Hayao Miyazaki",
        producer: "Isao Takahata",
        release_date: "1986",
        running_time: "124",
        rt_score: "95",
      };

      const movieCard = ReactTestRenderer.create(<MovieCard movie={movie} />);
      expect(movieCard.toJSON()).toMatchSnapshot();
    });
  });
});
