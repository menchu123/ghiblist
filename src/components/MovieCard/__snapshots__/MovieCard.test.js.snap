// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a MovieCard component When it is passed a movie object Then it should render a card matching the snapshot 1`] = `
<a
  href="/undefined"
  onClick={[Function]}
>
  <li
    className="movie"
  >
    <section
      className="movie__image"
    >
      <img
        alt="Spirited Away movie poster"
        className="movie__poster"
        height="300"
      />
    </section>
    <section
      className="movie__info"
    >
      <h3
        className="movie__title"
      >
        Spirited Away
      </h3>
      <div
        className="movie__data"
      >
        <p
          className="movie__year"
        >
          1986
        </p>
        <p
          className="movie__runtime"
        >
          124
          min
        </p>
        <p
          className="movie__score"
        >
          95
          %
        </p>
      </div>
    </section>
  </li>
</a>
`;
